macros/utils/data_types/data_type.sql

```diff
@@ -123,9 +123,5 @@
 {% endmacro %}
 
 {% macro default__edr_type_date() %}
```

Comment:
actually yes - I've noticed we're not handling dates correctly
I think we just didn't care about it, and Maayan needed to handle an edge case for BQ.
But I checked across all the warehouses we support and the correct date type for all of them is `date`.

I changed it because otherwise it caused filters like `metric_date in ('2023-11-01', '2023-11-03')` (which I added in my tests) not to work because the code cast them to a timestamp